crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -673,15 +676,25 @@ impl ScheduleGraph {
             }
         } else {
             for set in set_iter {
-                self.configure_set_inner(set).unwrap();
+                if let Err(e) = self.configure_set_inner(set) {
+                    // using `unwrap_or_else(panic!)` led to the error being reported
+                    // from this line instead of in the user code
```

Comment:
Might be worth opening a rust issue about this, that doesn't sound like intended behavior.

Reply:
https://github.com/rust-lang/rust/issues/115302